src/test/javascript/spec/component/programming-exercise/programming-exercise-instructor-submission-state.component.spec.ts

```diff
@@ -51,11 +51,21 @@ describe('ProgrammingExerciseInstructorSubmissionStateComponent', () => {
         return debugElement.query(By.css(resultEtaId));
     };
 
-    beforeEach(async () => {
+    beforeEach(() => {
         return TestBed.configureTestingModule({
-            imports: [TranslateModule.forRoot(), ArtemisTestModule, ArtemisProgrammingExerciseActionsModule],
+            imports: [ArtemisTestModule],
+            declarations: [
+                ProgrammingExerciseInstructorSubmissionStateComponent,
+                ProgrammingExerciseTriggerAllButtonComponent,
+                ButtonComponent,
+                FeatureToggleDirective,
+                FeatureToggleLinkDirective,
+                TranslatePipeMock,
+                MockDirective(NgbTooltip),
+                MockDirective(TranslateDirective),
+                MockPipe(DurationPipe),
+            ],
             providers: [
-                JhiLanguageHelper,
                 { provide: ParticipationWebsocketService, useClass: MockParticipationWebsocketService },
```

Comment:
Looking at the component, it only has a reference to the ProgrammingSubmissionService. Mocking only that would probably also suffice, unless the tests needs some functionality from there that cannot be mocked.

Reply:
You are right, I removed the ones which could be removed